app/portainer/services/async.js

```diff
@@ -0,0 +1,17 @@
+angular.module('portainer').factory('$async', ['$q',
```

Comment:
Might be worth inserting a comment in this file pointing to this PR for more documentation about this.